bumble/hci.py

```diff
@@ -5853,12 +6045,92 @@ def __str__(self) -> str:
         return (
             f'{color("SCO", "blue")}: '
             f'handle=0x{self.connection_handle:04x}, '
-            f'ps={self.packet_status}, rfu={self.rfu}, '
+            f'ps={self.packet_status}, '
             f'data_total_length={self.data_total_length}, '
             f'data={self.data.hex()}'
         )
 
 
+# -----------------------------------------------------------------------------
+class HCI_IsoDataPacket(HCI_Packet):
+    '''
+    See Bluetooth spec @ 5.4.5 HCI ISO Data Packets
+    '''
+
+    hci_packet_type = HCI_ISO_DATA_PACKET
+
+    @staticmethod
+    def from_bytes(packet: bytes) -> HCI_IsoDataPacket:
+        h = int.from_bytes(packet[0:2], 'little')
+        connection_handle = h & 0xFFF
+        pb_flag = (h >> 12) & 0x0F
+        ts_flag = (h >> 12) & 0x20
+        data_total_length = int.from_bytes(packet[2:4], 'little')
+        time_stamp = int.from_bytes(packet[4:8], 'little')
+        packet_sequence_number = int.from_bytes(packet[8:10], 'little')
+        h = int.from_bytes(packet[10:12], 'little')
+        iso_sdu_length = h & 0xFFF
+        packet_status_flag = h >> 14
+        iso_sdu_fragment = packet[12:]
+        return HCI_IsoDataPacket(
+            connection_handle=connection_handle,
+            pb_flag=pb_flag,
+            ts_flag=ts_flag,
+            data_total_length=data_total_length,
+            time_stamp=time_stamp,
+            packet_sequence_number=packet_sequence_number,
+            iso_sdu_length=iso_sdu_length,
+            packet_status_flag=packet_status_flag,
+            iso_sdu_fragment=iso_sdu_fragment,
+        )
+
+    def __init__(
+        self,
+        connection_handle: int,
+        pb_flag: int,
+        ts_flag: int,
+        data_total_length: int,
+        time_stamp: int,
+        packet_sequence_number: int,
+        iso_sdu_length: int,
+        packet_status_flag: int,
+        iso_sdu_fragment: bytes,
+    ) -> None:
+        self.connection_handle = connection_handle
+        self.pb_flag = pb_flag
+        self.ts_flag = ts_flag
+        self.data_total_length = data_total_length
+        self.time_stamp = time_stamp
+        self.packet_sequence_number = packet_sequence_number
+        self.iso_sdu_length = iso_sdu_length
+        self.packet_status_flag = packet_status_flag
+        self.iso_sdu_fragment = iso_sdu_fragment
+
+    def __bytes__(self) -> bytes:
+        return self.to_bytes()
+
+    def to_bytes(self) -> bytes:
+        buf = bytearray(12 + self.iso_sdu_length)
+        buf[0] = self.ts_flag << 14 | self.pb_flag << 12 | self.connection_handle
```

Comment:
Those are the first 2 bytes, not the first byte.
(same comment for other fields below)

Reply:
Done